src/v/storage/spill_key_index.cc

```diff
@@ -88,14 +88,34 @@ ss::future<> spill_key_index::add_key(compaction_key b, value_type v) {
     auto const key_size = b.size();
     auto const expected_size = idx_mem_usage() + _keys_mem_usage + key_size;
 
-    // TODO call into storage_resources
+    auto take_result = _resources.compaction_index_take_bytes(key_size);
```

Comment:
are we only account for the keys that are stored in the index, should we also account at least for the values related with them ?

Reply:
I was following the existing accounting's behavior of ignoring value sizes, but the including value size in the calculation is pretty simple, so we might as well do it -- I've added a commit.

Reply:
we do account for the values in `idx_mem_usage()`

Reply:
It looked to me like `AllocatedByteSize` on node_hash_map is just returning the space used for table's slots, and that the key+value were both allocated outside of the slot?